src/selectedtests/test_mappings/cli.py

```diff
@@ -0,0 +1,191 @@
+)
+@click.option(
+    "--module-name",
+    type=str,
+    help="The name of the associated module that should be analyzed. Example: enterprise",
+)
+@click.option(
+    "--module-source-file-regex",
+    type=str,
+    help="Regex that will be used to map what module files mappings will be created. "
+    "Example: 'src.*'",
+)
+@click.option("--module-test-file-regex", type=str, help="Regex to match test files in module.")
+@click.option(
+    "--output-file",
+    type=str,
+    help="Path to a file where the task mappings should be written to. Example: 'output.txt'",
+)
+def create(
+    ctx,
+    evergreen_project: str,
+    start: str,
+    end: str,
+    source_file_regex: str,
+    test_file_regex: str,
+    module_name: str,
+    module_source_file_regex: str,
+    module_test_file_regex: str,
+    output_file: str,
+):
+    """Create the test mappings for a given evergreen project."""
+    evg_api = ctx.obj["evg_api"]
+
+    try:
+        start_date = datetime.fromisoformat(start)
+        end_date = datetime.fromisoformat(end)
+    except ValueError as e:
+        LOGGER.error(str(e))
+        LOGGER.error("The start or end date could not be parsed - make sure it's an iso date")
+        return
+
+    evg_project = _get_evg_project(evg_api, evergreen_project)
+    repo_name = evg_project.repo_name
+    branch = evg_project.branch_name
+    repo_owner_name = evg_project.owner_name
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        project_repo = init_repo(temp_dir, repo_name, branch, repo_owner_name)
+        source_re = re.compile(source_file_regex)
+        test_re = re.compile(test_file_regex)
+        project_test_mappings = TestMappings.create_mappings(
+            project_repo, source_re, test_re, start_date, end_date, evergreen_project, branch
+        )
+        project_test_mappings_list = project_test_mappings.get_mappings()
+
+        if bool(module_name):
+            module_info = _get_module_info(evg_api, evergreen_project, module_name)
+            module_repo = init_repo(
+                temp_dir,
+                module_info["module_repo"],
+                module_info["module_branch"],
+                module_info["module_owner"],
+            )
+            module_source_re = re.compile(module_source_file_regex)
+            module_test_re = re.compile(module_test_file_regex)
+            module_test_mappings = TestMappings.create_mappings(
+                module_repo,
+                module_test_re,
+                module_source_re,
+                start_date,
+                end_date,
+                evergreen_project,
+                module_info["module_branch"],
+            )
+            module_test_mappings_list = module_test_mappings.get_mappings()
+            project_test_mappings_list.append(module_test_mappings_list)
+
+    json_dump = json.dumps(project_test_mappings_list, indent=4)
+
+    if output_file is not None and output_file != "":
+        f = open(output_file, "a")
```

Comment:
[nit] In python it is usually recommend to use `with` statements for opening files. This will automatically close the file when exiting the `with` so you don't need to worry about calling `close`.

```
with open(output_file, "a") as f:
    f.write(json_dump)
````